src/Action/Delivery/Create.php

```diff
@@ -26,7 +28,20 @@ public function __invoke(Delivery $data)
             throw new ValidationException($errors);
         }
 
-        $this->pricingManager->createOrder($data);
+        $useArbitraryPrice = $this->authorizationCheckerInterface->isGranted('ROLE_ADMIN') && !is_null($data->getDeliveryPriceInput());
```

Comment:
If you use `ArbitraryPrice` in `Delivery` class, `!is_null($data->getDeliveryPriceInput())` could be replaced by a more expressive & positive method name, `hasArbitraryPrice()`.